appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -631,22 +669,26 @@ Blockly.LexicalVariable.renameFree = function (block, freeSubstitution) {
   }
 }
 
+// TODO: Define what a free variable is.
```

Comment:
Conceptually, a free variable is one that hasn't be defined/bound yet (i.e., unbound). For example, if I say `var x = 5;`, `x` is not a free variable. However, if I write the expression `y + 2`, `y` is free because it hasn't been given any value.

For example, to create a free variable in App Inventor, you can create a variable with a local scope and then move its setter/getter blocks into a different scope so that the nam is no longer bound (i.e., it's free). This is visually represented in App Inventor by an error because when an unbound variable is encountered during execution it will just cause an error to be thrown.

Reply:
Makes sense! That shouldn't be too tricky to write a test for.